### PR TITLE
Correct the existing documented T-API functions to match the doxygen format

### PR DESCRIPTION
**Pull request by krishraghuram**
resolves part of #7197

### This pullrequest changes

Corrected the existing documented T-API functions to match the doxygen format

### CODE REVIEW

File: modules/core/include/opencv2/core/ocl.hpp [hunk 1]
```diff
@@ -278,55 +278,52 @@ class CV_EXPORTS Platform
     Impl* p;
 };
 
-/*
-//! @brief Attaches OpenCL context to OpenCV
+/**
+// @brief Attaches OpenCL context to OpenCV
```
Comment (mshabunin) [inline modules/core/include/opencv2/core/ocl.hpp]: There is no need in `//` at the beginning of each line of multi-line comment, please use one of the following formats:
```.cpp
/**
  .....
  .....
  .....
*/
```
or
```.cpp
//! ...
//! ...
//! ...
//! ...
```
Usually we use the first variant: [example](http://docs.opencv.org/master/d4/db1/tutorial_documentation.html#tutorial_documentation_quick_start_2).

Comment (alalek) [inline modules/core/include/opencv2/core/ocl.hpp]: Full list of doxygen supported comments is here (for reference): https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html

File: modules/core/include/opencv2/core/ocl.hpp [hunk 2]
```diff
@@ -569,7 +560,7 @@ class CV_EXPORTS Kernel
         i = set(i, a6); i = set(i, a7); i = set(i, a8); i = set(i, a9); i = set(i, a10); i = set(i, a11);
         i = set(i, a12); i = set(i, a13); i = set(i, a14); set(i, a15); return *this;
     }
-    /*
+    /**
     Run the OpenCL kernel.
```
Comment (mshabunin) [inline modules/core/include/opencv2/core/ocl.hpp]: This line can be used as a brief description (add `@brief` at the beginning).

File: modules/core/include/opencv2/core/ocl.hpp [hunk 3]
```diff
@@ -278,55 +278,46 @@ class CV_EXPORTS Platform
...
+@brief Attaches OpenCL context to OpenCV
+@note
+  OpenCV will check if available OpenCL platform has platformName name,
+  then assign context to OpenCV and call clRetainContext function.
+  The deviceID device will be used as target device and new command queue
+  will be created.
+@param platformName - name of OpenCL platform to attach,
+                     this string is used to check if platform is available
+                     to OpenCV at runtime
+@param platformID   - ID of platform attached context was created for
```
Comment (mshabunin) [inline modules/core/include/opencv2/core/ocl.hpp]: It is better to remove `-` symbol and manual alignment in all parameter descriptions:
```.cpp
/**
...
@param platformName name of OpenCL platform to attach, this string is used to check if platform is available to OpenCV at runtime
@param platformID ID of platform attached context was created for
@param context OpenCL context to be attached to OpenCV
@param deviceID ID of device, must be created from attached context
...
*/
```

File: modules/core/include/opencv2/core/ocl.hpp (hunk 1 above)
Comment (krishraghuram) [inline modules/core/include/opencv2/core/ocl.hpp]: I did notice the extra ```//!``` in the comments, but left it as is, assuming its a opencv coding style.
@mshabunin Thanks for the [example](http://docs.opencv.org/master/d4/db1/tutorial_documentation.html#tutorial_documentation_quick_start_2)

**Comment by krishraghuram** [comment]
Dumb question : 
why does it say "krishraghuram committed with sovrasov" in the first commit of this PR?

**Comment by mshabunin** [comment]
@krishraghuram , because @sovrasov has squashed your commits into one and this new commit has you recorded as the author and him as committer. You can also note that we've added two more commits to your branch with minor fixes, so now this PR is ready to be merged.
:+1:

**Pull request merged**